_posts/2020-07-08-pull-requests-require-context.md

```diff
@@ -0,0 +1,148 @@
+---
+layout: epic
+title: "Pull Requests Need Context"
+date: "2020-07-08"
+author: [steve-hicks]
+categories: [tools, github, team, community, engineering]
+---
+
+Code review is an engineering process that has benefited greatly from a move toward asynchronous communication.
+Long ago, engineering teams would sit around in a room together with code on a projector to review changes. 😱 For
+many teams this would mean batching code reviews or even skipping them altogether. 😱😱
+
+Today, most engineering teams use incredible tools like GitHub or GitLab to review changes through Pull Requests
+(PRs). The greatest advantage of PRs is that the review can happen when it's convenient for the reviewer:
+asynchronously.
+
+<!-- more -->
+
+One important feature _missing_ from asynchronous communication is the ability to course-correct when context is
+misunderstood. When you're in a synchronous conversation with someone, it doesn't take much time for them to let
+you know you've forgotten to include context. It doesn't take much time for you to add the missing context to keep
+the conversation moving forward.
+
+It takes a lot longer to identify missing context when communicating asynchronously. Worse, lack of context when
+_reviewing code_ asynchronously has a reverb effect. I create my PR when it's convenient for me, you ask a
```

Comment:
I like this reverb metaphor. I've been thinking about roundtrips over a slow 2G network, but this works way better.